javaslang/src/test/java/javaslang/concurrent/FutureTest.java

```diff
@@ -658,6 +659,65 @@ public void shouldThrowOnGetAfterCancellation() {
         fail("Future was expected to throw on get() after cancellation!");
     }
 
+    // -- collect()
+
+    @Test
+    public void shouldCollectDefinedValueUsingPartialFunction() {
+        final PartialFunction<Integer, String> pf = new PartialFunction<Integer, String>() {
+            @Override
+            public String apply(Integer i) {
+                return String.valueOf(i);
+            }
+            @Override
+            public boolean isDefinedAt(Integer i) {
+                return i % 2 == 1;
+            }
+        };
```

Comment:
Please reduce the duplicate code in all tests

```diff
+        final PartialFunction<Integer, String> pf = new PartialFunction<Integer, String>() {
+            @Override
+            public String apply(Integer i) {
+                return String.valueOf(i);
+            }
+            @Override
+            public boolean isDefinedAt(Integer i) {
+                return i % 2 == 1;
+            }
+        };
```

with this more concise variant:

```java
final PartialFunction<Integer, String> pf = Function1.<Integer, String> of(String::valueOf).partial(i -> i % 2 == 1);
```

(or `Function1.of(String::valueOf).partial(i -> i % 2 == 1);` ???)